File: assignment 10 - C# practice problems/34.cs
//Write a c# program to remove duplicate items from an array

using System;
using System.Collections.Generic;

namespace MyNameSpace
{
    class Program34
    {
        public static void RemoveDuplicatesDriverFunction()
        {

            var numArray = new List<int>() { 1, 2, 2, 5, 5, 5, 5, 5, 5, 5, 8, 7, 9, 9, 5, 9, 6 };
            Console.WriteLine("array is " + "[{0}]", string.Join(",", numArray));

            var resultArray = new List<int>();

            foreach (var item in numArray)
            {
                var result = resultArray.IndexOf(item);
                if (result == -1) resultArray.Add(item);

            }

            Console.WriteLine("array after removing duplicate is:" + "[{0}]", string.Join(",", resultArray));

        }



    }
}

File: assignment 3 - js practice problems/web-dev-starter/34.js
//Write a JavaScript program to remove duplicate items from an array
function removeDuplicates(arrayParam) {

    let uniqueArray = [];
    arrayParam.forEach((item) => {

        let itemFound = uniqueArray.findIndex((uniqueItem) => {
            return uniqueItem === item;
        })

        if (itemFound == -1) {
            uniqueArray.push(item);
        }
    })

    return uniqueArray;
}

function removeDuplicatesDriverFunction() {
    let arrayElement = [1, 2, 2, 5, 5, 5, 5, 5, 5, 5, 8, 7, 9, 9, 5, 9, 6];
    console.log("array is: " + arrayElement);
    console.log("array after removing duplicate is: " + removeDuplicates(arrayElement));
}

removeDuplicatesDriverFunction();